src/TurboMqtt/Client/ClientStreamOwner.cs

```diff
@@ -316,7 +316,7 @@ private void Running(object message)
             {
                 _log.Info("Server disconnected the client. Reason: {0}", serverDisconnect.Reason);
                 _log.Info("Client has exhausted all reconnect attempts. Shutting down.");
-                Context.Stop(Self);
+                Self.Tell(PoisonPill.Instance);
```

Comment:
Calling `Context.Stop` here was part of the problem - meant that we bailed out early and weren't able to complete a bunch of the client disconnect steps, leaving the `ConnectionStatus` in a corrupted state - this change is probably not necessary though. The real fix was in the `PostStop`.